solc/CommandLineParser.cpp

```diff
@@ -718,7 +718,7 @@ General Information)").c_str(),
 		(CompilerOutputs::componentName(&CompilerOutputs::signatureHashes).c_str(), "Function signature hashes of the contracts.")
 		(CompilerOutputs::componentName(&CompilerOutputs::natspecUser).c_str(), "Natspec user documentation of all contracts.")
 		(CompilerOutputs::componentName(&CompilerOutputs::natspecDev).c_str(), "Natspec developer documentation of all contracts.")
-		(CompilerOutputs::componentName(&CompilerOutputs::metadata).c_str(), "Combined Metadata JSON whose Swarm hash is stored on-chain.")
+		(CompilerOutputs::componentName(&CompilerOutputs::metadata).c_str(), "Request the combined Metadata JSON whose Swarm hash is stored on-chain. If used with --output-dir the metadata will be written in that directory, one file per contract each named \"<ContractName>_meta.json\", otherwise it is displayed on stdout.")
```

Comment:
> If used with --output-dir the metadata will be written in that directory, one file per contract each named "<ContractName>_meta.json", otherwise it is displayed on stdout.

This is true but it's how pretty much every output option works (and some others too like `--combined-json`). Mentioning it only for `--metadata` will give an impression that it's the only one that works this way.

Maybe it would be better to expand the description of `--output-dir` instead? Or, alternatively we could have some text for the whole "Output Components" section. IIRC each section can have its own description.

Reply:
I think the last suggestion is better then, putting it in the section header

Reply:
Turns out that section headers can't have more description than their caption.. at least I found no method to add text..

Reply:
Hmm... I guess we could hack that in by making the caption multi-line. Not sure if that would work out nice though.